### DISCUSSION TITLE
Is there a function that provides m-separation in partial ancestral graphs?

### DISCUSSION THREAD

**Discussion by kenneth-lee-ch** [agrumery/aGrUM#23] [Q&A] [answered]
Is there a function in pyagrum that provides m-separation in partial ancestral graphs described in this paper?: https://www.sciencedirect.com/science/article/pii/S0004370208001008?

**Comment by phwuil** [comment, marked as answer]
Hi @kenneth-lee-ch , not for now. It would be easy to implement but the main problem is to choose which concept (and terminology) has to be implemented.

For now, we have
- DAG and dseparation
- UndiGraph and separation
- PDAG and c-separation

- In the "nightly" version, pyAgrum includes a `MeekRules` class that completes a `MixedGraph` into a `PDAG` and/or `DAG`.

Now, I am quite interested in a complete proposition (yours ?) of which concepts, terminology and algorithms we should implement in order to be quite exhaustive in this domain.  I'm sure that if we were clear on this proposal, implementation would be swift.
